controllers/userStatus.js

```diff
@@ -84,28 +92,36 @@ const getAllUserStatus = async (req, res) => {
  */
 const updateUserStatus = async (req, res) => {
   try {
-    const { userId } = req.params;
-    const dataToUpdate = req.body;
-    const updateStatus = await userStatusModel.updateUserStatus(userId, dataToUpdate);
-    const { userStatusExists, userStatusUpdated, id, data } = updateStatus;
-    let responseObject = { id, userId };
-    let statusCode;
-    if (data) {
-      responseObject = { ...responseObject, ...data };
-    }
-    if (userStatusExists) {
-      responseObject.message = "User Status updated successfully.";
-      statusCode = 200;
+    let userId;
+    if (req.route.path === "/self") {
+      userId = req.userData.id;
     } else {
-      if (userStatusUpdated) {
-        responseObject.message = "User Status created successfully.";
-        statusCode = 201;
+      userId = req.params.userId;
+    }
```

Comment:
I see that this is being repeated in multiple controllers, would it be better to have a `getUserId` function that implements this logic and returns us the user id?